engine/userdata/src/main/java/org/apache/cloudstack/userdata/UserDataManagerImpl.java

```diff
@@ -90,49 +90,50 @@ public String concatenateUserData(String userdata1, String userdata2, String use
 
     @Override
     public String validateUserData(String userData, BaseCmd.HTTPMethod httpmethod) {
-        byte[] decodedUserData = null;
-        if (userData != null) {
-
-            if (userData.contains("%")) {
-                try {
-                    userData = URLDecoder.decode(userData, "UTF-8");
-                } catch (UnsupportedEncodingException e) {
-                    throw new InvalidParameterValueException("Url decoding of userdata failed.");
-                }
-            }
+        if (StringUtils.isBlank(userData)) {
```

Comment:
why is userdata obligatory?

Reply:
> why is userdata obligatory?

userdata is not mandatory, updated it to return null as earlier.